standalone/src/test/java/io/pravega/local/AuthEnabledInProcPravegaClusterTest.java

```diff
@@ -30,34 +29,17 @@
  * in the parent class.
  */
 @Slf4j
-public class AuthEnabledInProcPravegaClusterTest extends InProcPravegaClusterTest {
-    @Before
-    @Override
-    public void setUp() throws Exception {
-        this.authEnabled = true;
-        this.tlsEnabled = false;
-        super.setUp();
-    }
-
-    @Override
-    String scopeName() {
-        return "AuthTestScope";
-    }
+public class AuthEnabledInProcPravegaClusterTest {
 
-    @Override
-    String streamName() {
-        return "AuthTestStream";
-    }
-
-    @Override
-    String eventMessage() {
-        return "Test message on the plaintext channel with auth credentials";
-    }
+    @ClassRule
+    public static final PravegaEmulatorResource EMULATOR = new PravegaEmulatorResource(true, false);
```

Comment:
I'm curious how this thing works. I don't like it because it's static. How would this behave if we enabled parallel test runs (each class test suite may run in parallel with another)?

Reply:
This is similar to the @BeforeClass annotation, hence it is a static.  Also, i verified that `--parallel` option in gradle does not execute test methods inside a single test file in parallel.

Reply:
Confirmed. This is fine, provided it's shutdown method actually cleans everything up and frees all ports etc.

Reply:
Thanks for the clarification @shrids

Reply:
can we try BeforeClass

Reply:
The main benefit of junit rule is you have to write it once and use it as many times as you want. whereas setup() and tearDown() are the methods that have to be written for every TestClass on an ad-hoc basis.